pkg/controller/versions/strategy_v2_4.go

```diff
@@ -274,10 +274,41 @@ func (v *versionStrategyV2_4) ValidateRequest(ctx context.Context, cl client.Cli
 			return admission.ValidationResponse(false, "a cluster-scoped SMCP may only be created by users with cluster-admin permissions")
 		}
 	}
-
+	hasPilotSecretName := ContainsPilotSecretNameField(smcp)
+	if hasPilotSecretName {
+		return admission.ValidationResponse(false, "SMCP does not allow PilotSecretName field")
```

Comment:
Try to put yourself into the user's shoes. When they get this error message, will they know what's wrong? I think the user might then ask "why doesn't the SMCP allow this field?". It's best if the message itself explains why.

Reply:
I am having a hard time coming up with a more intellect response in this field. Any idea on what to say?

Reply:
Oh, I re-read the Jira issue and it looks like we shouldn't return an error but a deprecation warning. So, the message could read something like `spec.foo.bar.baz.pilotSecretName is deprecated; will be removed in 2.5.0; use foo.bar.baz.xyz instead`. 

We need to figure out how to emit a warning.

Reply:
> Oh, I re-read the Jira issue and it looks like we shouldn't return an error but a deprecation warning.

I can see in the JIRA issue "...if the deprecated fields exist, a warning should be returned."

Reply:
Yes, as I said: we must return a warning not an error.  And we can't do that until we upgrade k8s.io/api to v0.19+.

Reply:
Ah, ok, nevermind. I misunderstood your comment and I thought you mean I specified wrong requirements.